llama_stack/distribution/server/auth_providers.py

```diff
@@ -92,12 +100,16 @@ async def close(self):
         pass
 
 
+class KubernetesAuthProviderConfig(BaseModel):
+    api_server_url: str
+    ca_cert_path: str | None = None
+
+
 class KubernetesAuthProvider(AuthProvider):
```

Comment:
Kubernetes must have a standard OAuth setup too and so it is possible that we can replace this provider with the JWT one. But I am not very knowledgeable in these matters... @leseb ?

Reply:
Kubernetes does expose a jwks endpoint

Reply:
@grs I think in that case I propose that we should remove the existing K8S auth provider. @leseb also tells me it is not yet used.

Reply:
+1 for removing the K8s provider as it can be used only in very specific use cases (i.e., when the LLS API user also has access to the K8s API, which is not a typical setup)

Reply:
Yes, I think so. 

What I meant, is that a LLS API user typically is not exposed to the underlying infrastructure but only to the API endpoints. 

I'm talking here about production setups, not about developer setup (but in which case, authentication itself probably don't play much of a role). I think we should be overall clear what is targeting which persona. Also, the MLOps persona has obviously access to the K8s API server to manage LLS stack installations, but they are not the LLS API endusers. 

In other words, I don't believe that Llama Stack API users are also Kubernetes API users (in the most of the cases)

Reply:
The current kubernetes provider works with any serviceaccount token. For any process running in kubernetes that wants to access a llamastack it is simple to use the kubernetes provided token for a service account (which would be unique to that use and have no special permissions granted).
That said, I believe that this all still applies when using this new provider. It just requires specifying the jwks uri for the cluster in the config. So it would also make sense in my view to remove the kubernetes provider and use this one. (And of course alternative authz servers cuold then be used if desired).

Reply:
I'll take care of the removal and give spin at kube's jwks endpoint.